src/main/java/org/spongepowered/api/command/args/CommandContext.java

```diff
@@ -227,6 +249,26 @@ public boolean hasAny(Text key) {
         return hasAny(ArgUtils.textToArgKey(key));
     }
 
+    /**
+     * Returns whether the given flag has been defined in this context.
+     *
+     * @param key The key to look up
+     * @return whether the flag is defined
+     */
+    public boolean hasFlag(String key) {
```

Comment:
Do you actually need this new method?
AFIACT the hasAny method does the same (not limited to flags though).

Reply:
Yes, defined flags are stored in a set separate from the multimap used by arguments. This is because flags do not always have a value (or only one) and to avoid conflictions between flags and arguments with the same key. The split here was recommended by dual in the issue introducing this and I support the distinction as well.

Reply:
@ST-DDT See https://github.com/SpongePowered/SpongeAPI/issues/1978 for where the concern lies. There is a difference to "if a flag was specified" and "does it have a value".